crates/nu-command/src/formats/from/yaml.rs

```diff
@@ -174,6 +174,18 @@ fn convert_yaml_value_to_nu_value(
 
             Value::from(collected)
         }
+        serde_yaml::Value::Tagged(t) => {
+            let tag = &t.tag;
+            let value = match &t.value {
```

Comment:
I thought that the tagged value could be any value and not just string. Maybe we should support the others as well. What do you think?

Reply:
While fixing this issue I assumed that all tag values should be displayed as string.

However, I am learning more about [YAML](https://spacelift.io/blog/yaml).

It would be great if you can share few examples about tag types and how those should get displayed on nushell.

Reply:
Any more thoughts here @initinll?

Reply:
The way I see it is that any value having ! will match the below section - 

```rust

serde_yaml::Value::Tagged(t) => {
  let tag = &t.tag;
  let value = match &t.value {
      serde_yaml::Value::String(s) => s,
      _ => "",
  };
  
  Value::String {
      val: format!("{} {}", tag, value).trim().to_string(),
      span,
  }
}

```

So, any tag value should be represented as string, all the scenarios like tagged value with number, bool, string works, except I do not anticipate there will be any tagged mapping and sequence values in YAML, I did not find any example of that. 

Even the Serde_Yaml treats the tagged value as string, I have added few more passing tests to support this theory where test is to check number and bool types in tagged values.

Reply:
Again, good work on this. Thanks!

This is my thinking.
1. I'm fine landing with this as is because it's better than we have.
2. With this code
```rust
let value = match &t.value {
      serde_yaml::Value::String(s) => s,
      _ => "",
  };
```
`&t.value` could be any valid yaml type. It seems like we're only checking if it's a string and throwing the other types away.
3. Yaml can have tags like (i'm not sure these work):
- !!str just a string
- !!str true          # also just a string
- !!map
  a: 1
  b: 2
- !!seq
  - a
  - b
4. Yaml is confusing, especially this tagged value stuff.

Reply:
@fdncred I added the remaining cases, I think this covers all scenarios, what do you think?

```rust
serde_yaml::Value::Tagged(t) => {
  let tag = &t.tag;
  let value = match &t.value {
      serde_yaml::Value::String(s) => {
          let val = format!("{} {}", tag, s).trim().to_string();
          Value::String { val, span }
      }
      serde_yaml::Value::Number(n) => {
          let val = format!("{} {}", tag, n).trim().to_string();
          Value::String { val, span }
      }
      serde_yaml::Value::Bool(b) => {
          let val = format!("{} {}", tag, b).trim().to_string();
          Value::String { val, span }
      }
      serde_yaml::Value::Null => {
          let val = format!("{}", tag).trim().to_string();
          Value::String { val, span }
      }
      v => convert_yaml_value_to_nu_value(v, span, val_span)?,
  };
  
  value
}
```